packages/contracts-bedrock/test/Initializable.t.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ERC721Bridge_Initializer } from "test/CommonTest.t.sol";
+import { console } from "forge-std/console.sol";
+import { CrossDomainMessenger } from "src/universal/CrossDomainMessenger.sol";
+import { L2OutputOracle } from "src/L1/L2OutputOracle.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { ResourceMetering } from "src/L1/ResourceMetering.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+
+/// @title Initializer_Test
+/// @dev Ensures that the `initialize()` function on contracts cannot be called more than
+///      once. This contract inherits from `ERC721Bridge_Initializer` because it is the
+///      deepest contract in the inheritance chain for setting up the system contracts.
+contract Initializer_Test is ERC721Bridge_Initializer {
+    function test_cannotReinitializeL1_succeeds() public {
```

Comment:
Also worth asserting that the `L2Messenger` cannot be reinitialized.

Reply:
Thinking about how to handle this because really what we want to test is in practice these cannot be reinitialized. To do this, we would want to test against the L1 contracts being set up by the deploy script and test against the L2 contracts being set up by the genesis gen script

Reply:
i see. That sounds quite involved to add to this PR. We should look into doing this soon.